src/sql/workbench/browser/modal/modal.ts

```diff
@@ -416,7 +412,7 @@ export abstract class Modal extends Disposable implements IThemable {
 	 */
 	protected addFooterButton(label: string, onSelect: () => void, orientation: 'left' | 'right' = 'right'): Button {
 		let footerButton = DOM.$('.footer-button');
-		let button = new Button(footerButton);
+		let button = this._register(new Button(footerButton));
```

Comment:
you are adding the button to the disposable store, shouldn't we add the button click handler instead?

Reply:
This is registering it to the class level disposable store. When the dialog is disposed it is no longer usable so we want to dispose the entire button object.